src/pykeen/utils.py

```diff
@@ -990,14 +990,14 @@ def get_expected_norm(
         raise TypeError(f"norm not implemented for {type(p)}: {p}")
 
 
-def p_exp_map(v):
-    normv = torch.clamp(torch.norm(v, 2, dim=-1, keepdim=True), min=1e-10)
-    return torch.tanh(normv) * v / normv
+def p_exp_map(v: torch.FloatTensor, eps: float = 1.0e-10) -> torch.FloatTensor:
+    v_norm = v.norm(p=2, dim=-1, keepdim=True)
+    return v_norm.tanh() * v / v_norm.clamp_min(eps)
```

Comment:
is this because you don't need to clamp before running tanh()?

Reply:
As far as I saw it, the clamping was necessary to avoid division by zero. However, `v_norm` was also used as nominator, where there is no risk of numerical problems. So now, the clamping is only applied in the denominator.